queue/queue_service.js

```diff
@@ -6,6 +6,76 @@ queueApp.factory("QueueService", function($http){
 			},function(error){
 				return null;
 			});
+		},
+		checkInNow: function(time, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: 'action=checkIn&time=' + time + '&id=' + id,
```

Comment:
You probably should change this to be in JSON form:
```
{
  action: 'checkIn',
  time = time,
  id = id
}
```
I believe it's the prefer way to specifying it like this, because this is harder to read and change

Reply:
That's what I thought at first too. The problem here is actually that PHP can't handle Content-Type: application/json. We could write it as a JavaScript object, but then we'd have to override the Content-Type header and use a function to manually parse it into this same format. https://stackoverflow.com/a/20276775